File: projects/longmen-inn-system/backend/app/schemas/audit_log.py
"""
龙门客栈业务管理系统 - 审计日志 Schema
=====================================
作者: 厨子 (神厨小福贵)

用于审计日志API的数据模型定义
"""

from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel, Field


class AuditLogEntry(BaseModel):
    """审计日志条目"""
    timestamp: str = Field(..., description="审计时间")
    type: str = Field(default="git_audit", description="类型")
    files_changed: List[str] = Field(default_factory=list, description="变更文件列表")
    lines_added: int = Field(default=0, description="新增行数")
    lines_deleted: int = Field(default=0, description="删除行数")
    tasks_found: List[str] = Field(default_factory=list, description="关联任务ID")
    summary: str = Field(..., description="摘要")
    status: str = Field(default="passed", description="状态")
    issues: List[str] = Field(default_factory=list, description="问题列表")


class AuditLogResponse(BaseModel):
    """审计日志响应"""
    total: int
    entries: List[AuditLogEntry]
    statistics: dict


class AuditFeedResponse(BaseModel):
    """版本动态Feed响应"""
    feed: List[dict]

File: projects/longmen-inn-system/frontend/src/types/auditLog.ts
/**
 * 审计日志类型定义
 */

// 后端返回的审计日志条目
export interface AuditLogEntry {
  timestamp: string;
  type: string;
  files_changed: string[];
  lines_added: number;
  lines_deleted: number;
  tasks_found: string[];
  summary: string;
  status: string;
  issues: string[];
}

// 版本动态Feed条目（后端格式化后的格式）
export interface AuditFeedEntry {
  id?: string;
  timestamp: string;
  date_str?: string;
  title: string;
  status: string;
  status_emoji?: string;
  lines_added: number;
  lines_deleted: number;
  lines_info?: string;
  files_count: number;
  files_info?: string;
  issues: string[];
  issues_count: number;
  tasks_found: string[];
  type: string;
}

// 审计统计信息
export interface AuditStats {
  total_audits: number;
  total_files_changed: number;
  total_lines_added: number;
  total_lines_deleted: number;
  tasks_completed: number;
  issues_found: number;
}

// API响应类型
export interface AuditLogResponse {
  total: number;
  entries: AuditLogEntry[];
  statistics: Record<string, any>;
}

export interface AuditFeedResponse {
  feed: AuditFeedEntry[];
}
